sdk/core/azure-core/CMakeLists.txt

```diff
@@ -23,13 +23,35 @@ find_package(Threads REQUIRED)
 
 # min version for `CURLSSLOPT_NO_REVOKE`
 # https://curl.haxx.se/libcurl/c/CURLOPT_SSL_OPTIONS.html
-set(CURL_MIN_REQUIRED_VERSION 7.44)
+set(CURL_MIN_REQUIRED_VERSION 7.4)
 find_package(CURL ${CURL_MIN_REQUIRED_VERSION} CONFIG QUIET)
 if(NOT CURL_FOUND)
   find_package(CURL ${CURL_MIN_REQUIRED_VERSION} REQUIRED)
 endif()
 message("Libcurl version ${CURL_VERSION_STRING}")
 
+########################## Libcurl per-version found features. #################################
+#                                                                                              #
+#  Option `CURLSSLOPT_NO_REVOKE` will only be available for customers using version >= 7.44    #
+#                                                                                              #
+#  However, the Azure SDK for C++ can use an older version 7.4 witout supporting this option   #
+#                                                                                              #
+#    #     #              #            #           #          #            #                   #
+string(REPLACE "." ";" CURL_VERSION_LIST ${CURL_VERSION_STRING})
+list(GET CURL_VERSION_LIST 0 CURL_VERSION_MAJOR)
+list(GET CURL_VERSION_LIST 1 CURL_VERSION_MINOR)
+list(GET CURL_VERSION_LIST 2 CURL_VERSION_PATCH)
+set(OPTION_NO_REVOKE_MIN_MAJOR_VERSION "7")
+set(OPTION_NO_REVOKE_MIN_MINOR_VERSION "44")
+if(${CURL_VERSION_MAJOR} GREATER_EQUAL ${OPTION_NO_REVOKE_MIN_MAJOR_VERSION} AND ${CURL_VERSION_MINOR} GREATER_EQUAL ${OPTION_NO_REVOKE_MIN_MINOR_VERSION})
+  message("The libcurl version found supports the `CURLSSLOPT_NO_REVOKE` option. Building it as part of the transport adapter options.")
+  add_compile_definitions(BUILD_SSL_OPTION_NO_REVOKE)
```

Comment:
I think this is hitting the tipping point of how any compiler flags we expose to the user and this particular case doesn't seem common enough for us to add an SDK level option for.